lib/Macros/Sources/ObservationMacros/ObservableMacro.swift

```diff
@@ -303,14 +303,20 @@ public struct ObservationTrackedMacro: AccessorMacro {
       return []
     }
 
-    let initAccessor: AccessorDeclSyntax =
+    var accessors: [AccessorDeclSyntax] = []
+
+    if property.initializer == nil || !property.willSetAccessors.isEmpty || !property.didSetAccessors.isEmpty {
```

Comment:
The init accessor is required when willSetAccessor or didSetAccessor is present. [ObservableDidSetWillSet.swift](https://github.com/swiftlang/swift/blob/main/test/stdlib/Observation/ObservableDidSetWillSet.swift)